### ISSUE TITLE
关于灾难性遗忘有什么好办法吗？

### ISSUE THREAD

**Issue by ericjank** [shibing624/pycorrector#313]
### 基于以训练好的模型再继续训练时遇到了灾难性遗忘的问题

我们用30万训练集训练得到了不错的模型，结果比较满意，但是对于新增的词制作训练集并想再基于已经训练好的模型继续训练时遇到了灾难性遗忘的问题，尝试了锁定参数、调整学习率等方式都没有解决

**Comment by shibing624** [comment]
从头训练呢？

**Comment by ericjank** [comment]
> 

把新增的数据追加到训练集里从头练？需要花很长时间，主要是新增数据的频率很高，每次都从头练成本太高了 ：（

**Comment by shibing624** [comment]
1、新数据继续训练，lr减小10倍，batch size设置为8(或者2)，epoch设为3，参考continue training的方法和paper；
2、新的小数据集积累多批次，单独训练个模型，预测时跟大模型一起work。
3、我一般是从头训练，迭代模型 😄

**Comment by ericjank** [comment]
> 1、新数据继续训练，lr减小10倍，batch size设置为8(或者2)，epoch设为3，参考continue training的方法和paper；
> 2、新的小数据集积累多批次，单独训练个模型，预测时跟大模型一起work。
> 3、我一般是从头训练，迭代模型 😄

我试一下

**Comment by shibing624** [comment]
load optimizer and scheduler before continue training:
```
if os.path.isfile(os.path.join(args.model_name_or_path, "optimizer.pt")) and os.path.isfile(
            os.path.join(args.model_name_or_path, "scheduler.pt")):
        # Load in optimizer and scheduler states
        optimizer.load_state_dict(torch.load(os.path.join(args.model_name_or_path, "optimizer.pt")))
        scheduler.load_state_dict(torch.load(os.path.join(args.model_name_or_path, "scheduler.pt")))
```


save it after train, for next time load:
```
logger.info("Saving model checkpoint to %s", output_dir)
torch.save(optimizer.state_dict(), os.path.join(output_dir, "optimizer.pt"))
torch.save(scheduler.state_dict(), os.path.join(output_dir, "scheduler.pt"))
logger.info("Saving optimizer and scheduler states to %s", output_dir)
```

**Comment by ericjank** [comment]
> load optimizer and scheduler before continue training:
> 
> ```
> if os.path.isfile(os.path.join(args.model_name_or_path, "optimizer.pt")) and os.path.isfile(
>             os.path.join(args.model_name_or_path, "scheduler.pt")):
>         # Load in optimizer and scheduler states
>         optimizer.load_state_dict(torch.load(os.path.join(args.model_name_or_path, "optimizer.pt")))
>         scheduler.load_state_dict(torch.load(os.path.join(args.model_name_or_path, "scheduler.pt")))
> ```
> 
> save it after train, for next time load:
> 
> ```
> logger.info("Saving model checkpoint to %s", output_dir)
> torch.save(optimizer.state_dict(), os.path.join(output_dir, "optimizer.pt"))
> torch.save(scheduler.state_dict(), os.path.join(output_dir, "scheduler.pt"))
> logger.info("Saving optimizer and scheduler states to %s", output_dir)
> ```

好的 我再试试，得重新训练一天
